File: Hinweis Java/java/src/main/java/de/thm/mni/compilerbau/absyn/visitor/Visitor.java
package de.thm.mni.compilerbau.absyn.visitor;

import de.thm.mni.compilerbau.absyn.*;

/**
 * This interface is used to implement the visitor pattern.
 * <p>
 * You have to extend this class to implement your own visitor behavior.
 */
public interface Visitor {
    void visit(ArrayAccess arrayAccess);

    void visit(ArrayTypeExpression arrayTypeExpression);

    void visit(AssignStatement assignStatement);

    void visit(BinaryExpression binaryExpression);

    void visit(UnaryExpression unaryExpression);

    void visit(CallStatement callStatement);

    void visit(CompoundStatement compoundStatement);

    void visit(EmptyStatement emptyStatement);

    void visit(IfStatement ifStatement);

    void visit(IntLiteral intLiteral);

    void visit(NamedTypeExpression namedTypeExpression);

    void visit(NamedVariable namedVariable);

    void visit(ParameterDefinition parameterDefinition);

    void visit(ProcedureDefinition procedureDefinition);

    void visit(Program program);

    void visit(TypeDefinition typeDefinition);

    void visit(VariableDefinition variableDefinition);

    void visit(VariableExpression variableExpression);

    void visit(WhileStatement whileStatement);
}

File: Skeleton Kotlin/kotlin/src/main/kotlin/de/thm/mni/compilerbau/absyn/visitor/Visitor.kt
package de.thm.mni.compilerbau.absyn.visitor

import de.thm.mni.compilerbau.absyn.*

/**
 * This interface is used to implement the visitor pattern.
 *
 *
 * You have to extend this class to implement your own visitor behavior.
 */
interface Visitor {
    fun visit(arrayAccess: ArrayAccess)
    fun visit(arrayTypeExpression: ArrayTypeExpression)
    fun visit(assignStatement: AssignStatement)
    fun visit(binaryExpression: BinaryExpression)
    fun visit(unaryExpression: UnaryExpression)
    fun visit(callStatement: CallStatement)
    fun visit(compoundStatement: CompoundStatement)
    fun visit(emptyStatement: EmptyStatement)
    fun visit(ifStatement: IfStatement)
    fun visit(intLiteral: IntLiteral)
    fun visit(namedTypeExpression: NamedTypeExpression)
    fun visit(namedVariable: NamedVariable)
    fun visit(parameterDefinition: ParameterDefinition)
    fun visit(procedureDefinition: ProcedureDefinition)
    fun visit(program: Program)
    fun visit(typeDefinition: TypeDefinition)
    fun visit(variableDefinition: VariableDefinition)
    fun visit(variableExpression: VariableExpression)
    fun visit(whileStatement: WhileStatement)
}